### PR TITLE
#107 refactor tests

### PR DESCRIPTION
**Pull request by malte-b**
# Ticket
[#107](https://github.com/hpi-swa-teaching/SWT18-Project-08/issues/107)

# Description
Makes tests prettier.

# Is everything tested?
Yes.

# Is it release relevant?
No, internal only.

# coding style checks
 - [x] no external ressources needed
 - [x] coding standards 
    - [x] no `.` at the end of your function
    - [x] an empty line between functionname and code (except accessors)
    - [x] cascades wherever possible
    - [x] space around `@`, binary operators such as `+` or `<=` and after `^`
    - [x] no `or` stands alone in a line 
    - [x] variable definitions...
      - [x] ... have an space before/after the `|` symbol
      - [x] ... have a blank line above and under
      - [x] ... are used as few as possible   
   - [x] static values are stored on class Side
   - [x] every function has a category
   - [x] don't use magic numbers (and strings)!
 - [ ] basic UX
   - [ ] UI is intuitive to use (tested by the reviewer! reject if not)
   - [ ] ...

 ```smalltalk
example: aValue
   
   | variable |
   
   variable := 5.
   self myObject 
       value: aValue + variable;
       color: Color red.
   self otherMethod
   ```

### CODE REVIEW

File: packages/RichText-Core.package/RTETextArea.class/instance/openColorPicker.st [hunk 1]
```diff
@@ -0,0 +1,9 @@
+text manipulation
+openColorPicker 
+	
+	self colorpicker: (NewColorPickerMorph  
+		on: self 
+		originalColor: (self text asTextMorph selectionColor) 
+		setColorSelector: #applyColor:).
+	(self colorpicker submorphs at: 2) delete.
```
Comment (adrianjost) [inline packages/RichText-Core.package/RTETextArea.class/instance/openColorPicker.st]: magic number? Please comment what 2 is referencing to.

File: packages/RichText-Tests.package/RTEAddComponentTest.class/instance/testDeleteButtonOfCoreObject.st [hunk 2]
```diff
@@ -3,6 +3,4 @@ testDeleteButtonOfCoreObject
 
 	component componentChosen: RTEImageComponent.
 	self click: (self menuItemOf: component coreObject Named: 'deleteComponent').
-	self assert: component submorphs size equals: 1.
-	self assert: component submorphs first equals: component menu.
-	self assert: component coreObject equals: nil.
\ No newline at end of file
+	self assert: ((component submorphs size = 1) and: (component submorphs first = component menu) and: (component coreObject = nil))
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEAddComponentTest.class/instance/testDeleteButtonOfCoreObject.st]: We should discuss with tom if it is better to habe multiple asserts or and them. I like the previous version more but we should vote I think.

File: packages/RichText-Tests.package/RTEHeaderComponentTest.class/instance/testH1LargerH2LargerH3.st [hunk 3]
```diff
@@ -1,18 +1,15 @@
...
+	1 to: 3 do: [:i|
+		self click: (self menuItemNamed: 'h', i).
+		currentFontSizes at: i put: self currentFontSize].
 	
-	self click: (self menuItemNamed: 'h3').
-	h3size := self currentFontSize.
-	
-	self assert: h1size > h2size.
-	self assert: h2size > h3size. 
+	self assert: (currentFontSizes at: 1) > (currentFontSizes at: 2).
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEHeaderComponentTest.class/instance/testH1LargerH2LargerH3.st]: idea: move the assert into the loop and test if the previous size is larger than the currentFontSize. Edge case would be the first element. If you have an idea for that I would prefer to change it because it's easier to extent. (but obviously not needed)

File: packages/RichText-Tests.package/RTEHeaderComponentTest.class/instance/testH1LargerH2LargerH3.st [hunk 4]
```diff
@@ -1,18 +1,15 @@
 testing
 testH1LargerH2LargerH3
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEHeaderComponentTest.class/instance/testH1LargerH2LargerH3.st]: maybe rename to `testHeaderFontsizeDifferences` or something similar? H1, H2, H3 seems to be a bit to explicit

File: packages/RichText-Tests.package/RTEHorizontalLayoutComponentTest.class/instance/testWrappingLayoutChanged.st [hunk 5]
```diff
@@ -1,12 +1,9 @@
...
 
-	| addComponent |
-
-	component addComponent: RTEAddComponent.
-	addComponent := component submorphs first.
+	self setUpInsertTests.
 	
-	self click: ((self menuItemOf: addComponent Named: 'insertButtons')  submorphNamed: 'insertComponents'). 
-	self click: ((self menuItemOf: addComponent Named: 'insertButtons') submorphNamed: 'insertComponentUp'). 
+	self click: ((self menuItemOf: firstComponent Named: 'insertButtons') submorphNamed: 'insertComponents');
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEHorizontalLayoutComponentTest.class/instance/testWrappingLayoutChanged.st]: insert newline after `self`. Tobias hate the style we use here
```
self
    doStuff;
    moreStuff
```

File: packages/RichText-Tests.package/RTEImageComponentTest.class/instance/testMenuAlwaysVisibleWhenNoFileChosen.st [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
+	self flag: #refactor.
 	component filename: nil.
 	component menu show.
-	self
-		mouseEnter: component;
-		assert: component menu visible;
-		mouseLeave: component;
-		assert: component menu visible
\ No newline at end of file
+	self mouseEnter: component;
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEImageComponentTest.class/instance/testMenuAlwaysVisibleWhenNoFileChosen.st]: `self` should be in a new line when using cascades

File: packages/RichText-Tests.package/RTEInputFieldTest.class/instance/testAllowedCharacterRegex.st [hunk 7]
```diff
@@ -2,6 +2,5 @@ testing
 testAllowedCharacterRegex
 
 	component allowedCharacterRegex: '[0-9]'.
-	self
-		typeAllCharacters;
-		assert: component text equals: '0123456789'
\ No newline at end of file
+	self typeAllCharacters;
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEInputFieldTest.class/instance/testAllowedCharacterRegex.st]: see above, self + newline

File: packages/RichText-Tests.package/RTEInputFieldTest.class/instance/testMaxInputLength.st [hunk 8]
```diff
@@ -2,6 +2,5 @@ testing
 testMaxInputLength
 
 	component maxInputLength: 2.
-	self 
-		typeAllCharacters; "try >= 36 Characters"
-		assert: component text size equals: 2
\ No newline at end of file
+	self typeAllCharacters;
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEInputFieldTest.class/instance/testMaxInputLength.st]: see above, self + newline

File: packages/RichText-Tests.package/RTETextComponentTest.class/instance/selectTextByMouse.st [hunk 9]
```diff
@@ -1,8 +1,8 @@
...
 
 	self textArea text: 'some not so long text'.
-	self mousePress: (self textArea) buttons: #(left).	
+	self mousePress: self textArea buttons: #(left).	
 	self textArea selectFrom: 5 to: 8.
-	self assert: (self textArea selectedText) equals: 'not'.
-	self mouseRelease: (self textArea) buttons: #(left)
\ No newline at end of file
+	self assert: self textArea selectedText = 'not'.
+	self mouseRelease: self textArea buttons: #(left)
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTETextComponentTest.class/instance/selectTextByMouse.st]: is a cascade here possible?

File: packages/RichText-Tests.package/RTETextComponentTest.class/instance/testFontSizes.st [hunk 10]
```diff
@@ -1,18 +1,19 @@
...
 
-	| fontSizeField |
+	| fontSizeField input |
 
 	self textArea text: 'test'.
 	self textArea selectAll.
 	self mouseRelease: self textArea buttons: #(left).
 	
-	fontSizeField := (self menuItemNamed: 'fontSizeField').
+	input := #($1 $2 $-1).
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTETextComponentTest.class/instance/testFontSizes.st]: what is this line doing? Please add a comment for me :D `-1` seems interesting

File: packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentDown.st [hunk 11]
```diff
@@ -0,0 +1,7 @@
+testing
+testInsertComponentDown
+
+	self setUpInsertTests.
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentDown.st]: cascading possible?

File: packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentUp.st [hunk 12]
```diff
@@ -0,0 +1,7 @@
+testing
+testInsertComponentUp
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentUp.st]: cascading possible?

File: packages/RichText-Tests.package/RichTextEditorTest.class/instance/testEditorGetSavedByButton.st [hunk 13]
```diff
@@ -1,8 +1,10 @@
 testing
 testEditorGetSavedByButton
 
-	self assert: component isSaved not.
-	self flag: #fix. "for any reason, the click bellow has no effect. Manually tested and it works"
+	self deny: component isSaved.
+	self flag: #fix. "for any reason, the click bellow has no effect. Manually tested and it works. " "Don't know either why it doesn't work, really weird..."
```
Comment (adrianjost) [inline packages/RichText-Tests.package/RichTextEditorTest.class/instance/testEditorGetSavedByButton.st]: we should ask tom I think

**Comment by malte-b** [comment]
Changed your suggestion, you may have a look at it again @adrianjost

File: packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentDown.st (hunk 11 above)
Comment (malte-b) [reply packages/RichText-Tests.package/RTEVerticalLayoutComponentTest.class/instance/testInsertComponentDown.st]: doesn't make much sense in this case

**Comment by adrianjost** [comment]
@malte-b could you please resolve the merge conflicts? I'm a bit scared of breaking something here :/

**Comment by malte-b** [comment]
coveralls shows helper methods from tests as uncovered even if they are used in tests. i think it doesn't make sense to test our tests so i leave it as is. conflicts are now resolved.

**Comment by adrianjost** [comment]
@malte-b that's wrong! I thought so too but I was wrong. Coveralls count helper methods as tested as long as they are called anywhere during any test. In this case this means these helper functions are never called. We should fix this.

**Comment by malte-b** [comment]
@adrianjost but that's simply not true as the methods in your screenshot are used in some tests
